dbld/builddeps

```diff
@@ -188,7 +188,7 @@ function install_pip_packages {
                 $python_executable -m pip install --no-cache-dir --upgrade "setuptools < 58.0.0" # Remove the version limit when funcparserlib v1.0.0 is released
                 filter_packages_by_platform $DBLD_DIR/pip_packages.manifest | xargs $python_executable -m pip install --no-cache-dir --ignore-installed -U
                 ;;
-            fedora-35)
+            fedora-*)
```

Comment:
This was not an issue in fedora-33, I hope it gets fixed and as such I would not wildcard it for all fedora.

Reply:
I am not sure. Pip refuses to upgrade a package that is installed from an rpm.

And fedora-35 does have those installed by default. I think they were installed even before that but in those cases pip wasn't so selective.

I think a long term solution would be to use a virtualenv instead of messing with the system python libs.

Reply:
Nope, it's not related to python package versions... Rebasing this PR on #3960 didn't help.
It should virtualenv in the long term...

```
Step 17/23 : RUN /dbld/builddeps install_pip_packages
 ---> Running in 62d3f8ab1e27
+ install_pip_packages
+ case "${IMAGE_PLATFORM}" in
+ python_executables=python3
+ for python_executable in ${python_executables}
+ case "${IMAGE_PLATFORM}" in
+ python3 -m pip install --no-cache-dir --upgrade pip
Requirement already satisfied: pip in /usr/lib/python3.10/site-packages (21.2.3)
Collecting pip
  Downloading pip-22.0.4-py3-none-any.whl (2.1 MB)
Installing collected packages: pip
Successfully installed pip-22.0.4
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
+ python3 -m pip install --no-cache-dir --upgrade setuptools
Requirement already satisfied: setuptools in /usr/lib/python3.10/site-packages (57.4.0)
Collecting setuptools
  Downloading setuptools-61.2.0-py3-none-any.whl (1.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 8.5 MB/s eta 0:00:00
Installing collected packages: setuptools
  Attempting uninstall: setuptools
    Found existing installation: setuptools 57.4.0
ERROR: Cannot uninstall setuptools 57.4.0, RECORD file not found. Hint: The package was installed by rpm.
The command '/bin/sh -c /dbld/builddeps install_pip_packages' returned a non-zero code: 1
make: *** [dbld/rules:217: image-fedora-35] Error 1
```

Reply:
I used the asterisk I expect us to move on to Fedora-35 and that newer releases of Fedora would fail similarly.

Reply:
I can let this go, the breaking github-actions workflow is more painful to me.
Sooner or later, old setuptools will lead to a failing pip install when one of the unversioned pip modules is updated.

I saw that in Fedora 33 the pip upgrade works, but it warns that upgrading a system-installed pip is dangerous:
```
[root@b9026149357f /]# python3 -m pip install --no-cache-dir --upgrade setuptools                                                   
Requirement already satisfied: setuptools in /usr/lib/python3.9/site-packages (49.1.3)                                                                                                                                                                                  
Collecting setuptools                                                                                                               
  Downloading setuptools-61.2.0-py3-none-any.whl (1.1 MB)                                                                                                                                                                                                               
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 4.9 MB/s eta 0:00:00                                                                                                                                                                                           
Installing collected packages: setuptools                                                                                           
  Attempting uninstall: setuptools                                                                                                  
    Found existing installation: setuptools 49.1.3                                                                                  
    Uninstalling setuptools-49.1.3:                                                                                                                                                                                                                                     
      Successfully uninstalled setuptools-49.1.3                                                                                                                                                                                                                        
Successfully installed setuptools-61.2.0                                                                                                                                                                                                                                
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
```

I vote that we should merge this after #3960, hence the vim-workaround patch can be dropped in this PR.